client/web/compose/src/components/PageBlocks/Navigation/Base.vue

```diff
@@ -0,0 +1,158 @@
+                  size="sm"
+                />
+              </span>
+            </b-button>
+
+            <b-popover
+              ref="dropdown-popover"
+              :target="`dropdown-popover-${index}`"
+              :placement="navItem.options.item.align"
+              delay="0"
+              boundary="window"
+              triggers="click blur"
+            >
+              <template
+                v-if="navItem.type === 'dropdown'"
+              >
+                <div
+                  v-for="(dropdown, dIndex) in navItem.options.item.dropdown.items"
+                  :key="`dropdown-${dIndex}`"
+                >
+                  <a
+                    class="dropdown-item"
+                    :href="dropdown.url"
+                    :disabled="navItem.options.disabled"
+                    :target="selectTargetOption(dropdown.target)"
+                    :style="{ order: dIndex * 2 }"
+                  >
+                    {{ dropdown.label }}
+                  </a>
+
+                  <hr
+                    v-if="dropdown.delimiter"
+                    class="my-1"
+                    :style="{ order: dIndex + 1 }"
+                  >
+                </div>
+              </template>
+
+              <template v-else>
+                <div
+                  v-for="(dropdown, dIndex) in getSubPages(navItem.options.item.pageID)"
+                  :key="`dropdown-${dIndex}`"
+                >
+                  <b-link
+                    class="dropdown-item"
+                    :to="{ name: 'page', params: { pageID: dropdown.pageID } }"
+                    :target="selectTargetOption(dropdown.target)"
+                    :style="{ order: dIndex * 2 }"
+                    :disabled="navItem.options.disabled"
+                  >
+                    {{ dropdown.title }}
+                  </b-link>
+                </div>
+              </template>
+            </b-popover>
+          </template>
+
+          <template v-else-if="!navItem.options.item.displaySubPages">
+            {{ navItem.options.item.label }}
+          </template>
+        </b-nav-item>
+      </b-nav>
+    </div>
+  </wrap>
+</template>
+<script>
+import { mapGetters } from 'vuex'
+import base from '../base'
+
+export default {
+  extends: base,
+
+  computed: {
+    ...mapGetters({
+      pages: 'page/set',
+    }),
+  },
+
+  methods: {
+    getSubPages (selfID) {
+      return this.pages.filter(value => value.selfID === selfID) || []
+    },
+
+    selectTargetOption (target) {
+      switch (target) {
+        case 'sameTab':
+          return '_self'
+        case 'newTab':
+          return '_blank'
+        case 'newWindow':
+          return '_blank'
+      }
+    },
+
+    displayDropdownText (navItem) {
+      if (navItem.type === 'dropdown') {
+        return navItem.options.item.dropdown.label
+      }
+
+      return navItem.options.item.label
+    },
+
+    toggleToLink (navItem) {
```

Comment:
all good, but can you not call it toggle, since that implies you're toggling the value.. 
It should be more like, getToLink, getHrefLink